e2e/cypress/support/api/system.js

```diff
@@ -105,3 +122,15 @@ Cypress.Commands.add('apiInvalidateCache', () => {
         cy.wrap(response);
     });
 });
+
+function uploadLicencseIfDoesNotExist() {
+    // # Upload a license if it does not exist.
+    cy.apiGetClientLicense().then(({license}) => {
+        if (license.IsLicensed === 'false') {
+            cy.apiUploadLicense('mattermost-license.txt');
+
+            // * Verify license exists via admin console
+            cy.uiCheckLicenseExists();
```

Comment:
I find this not following the approach we're trying to do with custom commands, especially for API commands where it should avoid mixing with UI calls.

Reply:
@saturninoabril Any suggestions on what the approach should be? I can just add a big timeout but I can't really test that it's failing because it's passing for me locally.

Reply:
I played around it earlier and came up with [something like this](https://github.com/saturninoabril/mattermost-cypress-docker/compare/file-check#diff-de80b964a5e58206daa24f418a3dc1d8R121-R138). Then look for usage at `apiRequireLicense` and `apiRequireLicenseForFeature`.

Reply:
@saturninoabril I'm not sure that resolves the issue regarding the timing that license is available in the ui but I like the additional check for the file if it exists locally. Can you please pull the latest and then commit your changes? Looks your changes are spread out and would be easier for you to push the changes.

Reply:
It's not dealing on timing since it's returning a resolve function.
Yup, will send a follow up PR that's why I didn't request for changes.

Reply:
What I meant to say is I'm good with this PR and can merge once approved. Then I'll submit a separate PR for another round of review. But I can push my changes here too if you prefer it. Let me know.

Reply:
I'd prefer the latter so Prapti and Rohitesh can review in one go

Reply:
Done - https://github.com/mattermost/mattermost-webapp/pull/6101/commits/4a589db862a91617db1cb828eae4ca4baf2a3907. Please run on your end and let me know if you find anything.

Reply:
@saturninoabril Thanks! I retested and spot checked several tests and they're passing. Failing when I remove the license from UI as well as fixture, which is expected